Mathlib/CategoryTheory/Whiskering.lean

```diff
@@ -245,6 +263,37 @@ instance isIso_whiskerRight {G H : C ⥤ D} (α : G ⟶ H) (F : D ⥤ E) [IsIso
     IsIso (whiskerRight α F) :=
   (isoWhiskerRight (asIso α) F).isIso_hom
 
+@[simp]
+theorem isoWhiskerLeft_trans (F : C ⥤ D) {G H K : D ⥤ E} (α : G ≅ H) (β : H ≅ K) :
+    isoWhiskerLeft F (α ≪≫ β) = isoWhiskerLeft F α ≪≫ isoWhiskerLeft F β :=
+  rfl
+
+theorem isoWhiskerLeft_trans_assoc (F : C ⥤ D) {G H K : D ⥤ E} (α : G ≅ H) (β : H ≅ K)
+    {K' : C ⥤ E} (γ : F ⋙ K ≅ K') :
+    isoWhiskerLeft F (α ≪≫ β) ≪≫ γ = isoWhiskerLeft F α ≪≫ isoWhiskerLeft F β ≪≫ γ := by
+  rw [← Iso.trans_assoc, isoWhiskerLeft_trans]
+
+@[simp]
+theorem isoWhiskerRight_trans {G H K : C ⥤ D} (α : G ≅ H) (β : H ≅ K) (F : D ⥤ E) :
+    isoWhiskerRight (α ≪≫ β) F = isoWhiskerRight α F ≪≫ isoWhiskerRight β F :=
+  ((whiskeringRight C D E).obj F).mapIso_trans α β
+
+theorem isoWhiskerRight_trans_assoc {G H K : C ⥤ D} (α : G ≅ H) (β : H ≅ K) (F : D ⥤ E)
```

Comment:
It would be very nice to generalize the `reassoc` attribute so that it works also for equalities between isomorphisms. I do not know enough about metacode to do this, but the code in `Tactic.CategoryTheory.Reassoc` mostly rely on the lemma `eq_whisker`. It should be possible to do the same for a similar lemma for isomorphisms. Could you investigate this? (and maybe seek help on Zulip).

Reply:
I agree that this would be nice. I also do not know enough metaprogramming to make this happen in short time, so I cannot give any ETA for this.

I can investigate, but I also have an upcoming series of PR that uses some of the lemmas in this one, and I would rather not have it blocked by meta stuff like this for this that I don’t know for sure that I can make happen in a reasonable time frame.

So for now I’ll just remove the `assoc` versions of the lemmas here, and we'll tag `@[reassoc]` the lemmas here if/when this extension of `reassoc` comes out.